### PR TITLE
java advanced is completed

### PR DESCRIPTION
**Pull request by Ivanenko16**

### CODE REVIEW

File: src/main/java/javaadvanced/task1/CharacterCount.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package javaadvanced.task1;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class CharacterCount {
+
+    static void characterCount(String inputString) {
```
Comment (ivanmatiashepam) [inline src/main/java/javaadvanced/task1/CharacterCount.java:8]: please change return type to Map according to the task

File: src/main/java/javaadvanced/task2/SortableCollection.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package javaadvanced.task2;
+
+import java.util.Collection;
+import java.util.Set;
+import java.util.TreeSet;
+
+public class SortableCollection {
+
+    static <V> void createSortableCollection(Collection<V> collection) {
```
Comment (ivanmatiashepam) [inline src/main/java/javaadvanced/task2/SortableCollection.java:9]: Please use T keyword for Type. Also, please return this collection

File: src/main/java/javaadvanced/task4/Run.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        orders[2] = "Order2";
+        orders[3] = "Order3";
+        orders[4] = "Order4";
+        orders[5] = "Order5";
+        orders[6] = "Order5";
+        orders[7] = "Order5";
+
+        System.out.println("List of HashSet:");
+        HashSet<String> orderHashSet = new HashSet<>(Arrays.asList(orders));
+        for (String str : orderHashSet) {
```
Comment (ivanmatiashepam) [inline src/main/java/javaadvanced/task4/Run.java:19]: btw, you can also print the whole collection (if you do not need each value to be printed on a new line) - System.out.println(orderHashSet);

File: src/main/java/javaadvanced/task5/Run.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+        Map<K, V> map = new HashMap<>();
+
+        Iterator<K> keyIterator = keys.iterator();
+        Iterator<V> valueIterator = values.iterator();
+
+        while (keyIterator.hasNext()) {
+            if (valueIterator.hasNext()) {
+                map.put(keyIterator.next(), valueIterator.next());
+            } else {
```
Comment (ivanmatiashepam) [inline src/main/java/javaadvanced/task5/Run.java:64]: we do not need this part of code according to the task. there shouldn't be such situation - if keys > values there would be an Exception

**Review by ivanmatiashepam — CHANGES_REQUESTED** [review]

File: src/main/java/javaadvanced/task1/CharacterCount.java (hunk 1 above)
Comment (Ivanenko16) [reply src/main/java/javaadvanced/task1/CharacterCount.java:8]: Type changed

File: src/main/java/javaadvanced/task2/SortableCollection.java (hunk 2 above)
Comment (Ivanenko16) [reply src/main/java/javaadvanced/task2/SortableCollection.java:9]: Fixed

File: src/main/java/javaadvanced/task4/Run.java (hunk 3 above)
Comment (Ivanenko16) [reply src/main/java/javaadvanced/task4/Run.java:19]: Absolutely agree), fixed

File: src/main/java/javaadvanced/task5/Run.java (hunk 4 above)
Comment (Ivanenko16) [reply src/main/java/javaadvanced/task5/Run.java:64]: Fixed

File: src/main/java/javaadvanced/task2/SortableCollection.java (hunk 2 above)
Comment (ivanmatiashepam) [reply src/main/java/javaadvanced/task2/SortableCollection.java]: method is still void

File: src/main/java/javaadvanced/task4/Run.java (hunk 3 above)
Comment (ivanmatiashepam) [reply src/main/java/javaadvanced/task4/Run.java]: the same can be applied to the orderTreeSet as well, to have some common code style

File: src/main/java/javaadvanced/task2/SortableCollection.java (hunk 2 above)
Comment (Ivanenko16) [reply src/main/java/javaadvanced/task2/SortableCollection.java]: The method changed (return collection)

File: src/main/java/javaadvanced/task4/Run.java (hunk 3 above)
Comment (Ivanenko16) [reply src/main/java/javaadvanced/task4/Run.java]: Applied to the orderTreeSet too

File: src/main/java/javaadvanced/task2/SortableCollection.java (hunk 2 above)
Comment (ivanmatiashepam) [reply src/main/java/javaadvanced/task2/SortableCollection.java]: Please provide the parametrized type for this generic. should be - "static <T> Collection<T> createSortableCollection" (return collection's members type should be the same as the type for method's argument). 

Generic types shouldn’t be used raw (without type parameters) in variable declarations or return values. Doing so bypasses generic type checking, and defers the catch of unsafe code to runtime.

Comment (Ivanenko16) [reply src/main/java/javaadvanced/task2/SortableCollection.java]: Fixed, thanks for the explanation

**Pull request merged**